.aegir.js

```diff
@@ -2,9 +2,21 @@
 
 const createServer = require('ipfsd-ctl').createServer
 
-const server = createServer()
+// TODO(victor) fix this mess
+// This is a horrible mess to be able to get a free port when running ports.
+// Karma seems involved in the communication somehow, and then we have a HAPI
+// server in the middle that I couldn't figure out how to get the resolved port
+// from.
+//
+// So we have this for now. It works, but it's not proper.
+const getPortSync = require('get-port-sync')
+const freePort = getPortSync()
+process.env.AEGIR_TEST_PORT = freePort
+const server = createServer({port: freePort})
```

Comment:
All of these changes (throughout the code) look messy. Can we make this in a separate PR?

Reply:
I want to get rid of this hack completely but got stuck trying to pass things from js-ipfs -> karma -> browser tests, hence this hack.

The hack is to get tests to pass, as currently the tests are using ports that are already occupied. Without this fix, the tests here will fail.

Reply:
`s/ports/port` :) 

What is the way to do this cleanly? This is adding lots of stuff in multiple places just to save us from selecting one port. I argue that having always the same test port is actually a better way because it tells us when things are not closed properly.

Reply:
Be able to pass things between controlled browsers and controller (karma in this case). 

I would argue having dynamic ports are much better, as if we want to run multiple instances of the tests, we can do so on the same machine without any worries.

Having a hard-coded port will make that impossible.

With that said, I agree, discovering lingering things will be harder. We could probably have some checks by checking process-list and open sockets before running tests, and doing the same check after running tests, doing a diff between them and warn if something didn't close properly.

Reply:
I believe that we are in the same page that we need https://github.com/ipfs/aegir/issues/201, given that, I prefer to not have a _"TODO(victor) fix this mess"_ in the code for one single port. Note, this is not solving the same issue for js-ipfs-api, interop or even js-ipfsd-ctl. We need a solution for all.

That said. Let's keep it as it was before (ipfsd-ctl was released with a non-colliding port) and implement the real fix in aegir -- https://github.com/ipfs/aegir/issues/201.

Reply:
Sure, I agree, we need #201 but let's not push this PR to do too much. There is no reason we cannot move forward with this and implement ipfs/aegir#201 later on.

The problem is not with the other repositories, the problem is with this repository, so I don't see why the fix would be applies to other repositories?